debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -178,20 +180,26 @@ void logStatistics(final List<SourceRecord> records) {
 
         if (batchSize > 0) {
             SourceRecord lastRecord = records.get(batchSize - 1);
-            lastOffset = lastRecord.sourceOffset();
+            updateLastOffset(lastRecord.sourcePartition(), lastRecord.sourceOffset());
             previousOutputBatchSize += batchSize;
             if (pollOutputDelay.hasElapsed()) {
                 // We want to record the status ...
                 final Instant currentTime = clock.currentTime();
                 LOGGER.info("{} records sent during previous {}, last recorded offset: {}", previousOutputBatchSize,
-                        Strings.duration(Duration.between(previousOutputInstant, currentTime).toMillis()), lastOffset);
+                        Strings.duration(Duration.between(previousOutputInstant, currentTime).toMillis()), lastRecord.sourceOffset());
```

Comment:
It probably makes sense to log partition here too.

Reply:
I added the partition to the log message.